webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/uploading/UploadService.scala

```diff
@@ -185,21 +199,24 @@ class UploadService @Inject()(dataSourceRepository: DataSourceRepository,
   def addUploadIdsToUnfinishedUploads(
       unfinishedUploadsWithoutIds: List[UnfinishedUpload]): Fox[List[UnfinishedUpload]] =
     for {
-      maybeUnfinishedUploads: List[Box[UnfinishedUpload]] <- Fox.sequence(
+      maybeUnfinishedUploads: List[Box[Option[UnfinishedUpload]]] <- Fox.sequence(
+        // Filter out unfinished uploads that do not have entries in the redis store and adds the uploadId and filePaths to those that have an entry.
```

Comment:
```suggestion
        // Previously rejected uploads may still appear in this list, but don’t have entries in redis. We can use that to filter them out here, since we don’t want to list them to the user. Those that *do* have entries in redis are then enriched using info from there (uploadId and filePaths).
```
(Is this correct? To me your version of the comment seemed like more a description of the code than of intent. What do you think?)

Reply:
Now that I think about this, it is not entirely correct.
> Previously rejected uploads may still appear in this list,

Should usually not be the case as the clean up upon failure of the upload also removes them from postgres (as far as I know). But if this fails (or is forgotten like I did at some stage of this PR), then the datasets would appear in the list.

So this is more like a save guard against missing / failing to delete the postgres entry. 

So what do you think: Is it ok to keep this check here that removes datasets that are not present in the redis as they cannot be resumed anyway? Or should this code be removed as this case should very rarely happen (e.g. wrong code / forgot to remove or failed rpc to the main backend)